File: app/src/main/java/com/ayat/candyapp/di/FactoryModule.java
package com.ayat.candyapp.di;

import androidx.lifecycle.ViewModelProvider;
import com.blessedtreeit.product.activitycenter.ViewModelProviderFactory;
import com.blessedtreeit.product.activitycenter.flows.add_task.AddTaskViewModel;
import com.blessedtreeit.product.activitycenter.flows.entry.EntryViewModel;
import com.blessedtreeit.product.activitycenter.flows.login.LoginViewModel;
import com.blessedtreeit.product.activitycenter.flows.nav_drawer.NavViewModel;
import com.blessedtreeit.product.activitycenter.flows.tasks.TasksViewModel;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Ayat khraisat  on 12/18/18
 * akhraisat@blessedtreeit.com
 * <p>
 * Project Name: PocMvp
 * <p>
 * Blessed Tree IT
 */

@Module
public class FactoryModule {

    public FactoryModule() {

    }

    @Provides
    @Qualifiers.LoginViewModel
    ViewModelProvider.Factory provideLoginViewModelFactory(LoginViewModel loginViewModel) {
        return new ViewModelProviderFactory<>(loginViewModel);
    }

    @Provides
    @Qualifiers.EntryViewModel
    ViewModelProvider.Factory provideEntryViewModelFactory(EntryViewModel entryViewModel) {
        return new ViewModelProviderFactory<>(entryViewModel);
    }

    @Provides
    @Qualifiers.TaskViewModel
    ViewModelProvider.Factory provideTasksViewModelFactory(TasksViewModel tasksViewModel) {
        return new ViewModelProviderFactory<>(tasksViewModel);
    }

    @Provides
    @Qualifiers.NavViewModel
    ViewModelProvider.Factory provideNavViewModelFactory(NavViewModel navViewModel) {
        return new ViewModelProviderFactory<>(navViewModel);
    }

    @Provides
    @Qualifiers.AddTaskViewModel
    ViewModelProvider.Factory provideAddTaskViewModelFactory(AddTaskViewModel addTaskViewModel) {
        return new ViewModelProviderFactory<>(addTaskViewModel);
    }

}

File: app/src/main/java/com/ayat/candyapp/di/FactoryModule.kt
package com.ayat.candyapp.di

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.ayat.candyapp.ViewModelFactory
import com.ayat.candyapp.login.LoginViewModel
import dagger.Binds
import dagger.Module
import dagger.Provides
import dagger.multibindings.IntoMap

/**
 *Created by Ayat Khriasat on 27,April,2019 at 8:32 PM
 *Email: ayatzkhraisat@gmail.com
 *Project: CandyApp
 **/
@Module
abstract class FactoryModule {

    @Binds
    @ViewScope
    abstract fun bindViewModelFactory(viewModelFactory: ViewModelFactory): ViewModelProvider.Factory

    @Binds
    @IntoMap
    @ViewScope
    @ViewModelKey(LoginViewModel::class)
    abstract fun bindMyViewModel(myViewModel: LoginViewModel): ViewModel

}
